src/EditorFeatures/Core/Shared/Extensions/ITextBufferEditExtensions.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.CodeAnalysis.Editor.Shared.Extensions
+{
+    internal static class ITextBufferEditExtensions
+    {
+        private static Exception s_lastException = null;
+
+        /// <summary>
+        /// Logs exceptions thrown during <see cref="ITextBufferEdit.Apply"/> as we look for issues.
+        /// </summary>
+        /// <param name="edit"></param>
+        /// <returns></returns>
+        public static ITextSnapshot ApplyAndLogExceptions(this ITextBufferEdit edit)
+        {
+            try
+            {
+                return edit.Apply();
+            }
+            catch (Exception e) when (ErrorReporting.FatalError.ReportWithoutCrash(e))
```

Comment:
Do we think the exception escapes to here?  I thought in @jmarolf's previous investigation, it was already caught by the editor?  Or did I misunderstand?